### PR TITLE
[Merged by Bors] - feat: expand/review API about `gauge`

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #5270

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5270
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5270~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Analysis/Convex/Gauge.lean [hunk 1]
```diff
@@ -445,25 +514,58 @@ theorem Seminorm.gaugeSeminorm_ball (p : Seminorm ℝ E) :
...
+theorem gauge_closure_zero : gauge (closure (0 : Set E)) = 0 := funext fun x ↦ by
+  simp only [← singleton_zero, gauge_def', mem_closure_zero_iff_norm, norm_smul, mul_eq_zero,
+    norm_eq_zero, inv_eq_zero]
+  rcases (norm_nonneg x).eq_or_gt with hx | hx
+  · convert csInf_Ioi (a := (0 : ℝ))
+    exact Set.ext fun r ↦ and_iff_left (.inr hx)
+  · convert Real.sInf_empty
+    exact eq_empty_of_forall_not_mem fun r ⟨hr₀, hr⟩ ↦ hx.ne' <| hr.resolve_left hr₀.out.ne'
+
+-- TODO: use continuity?
```
Comment (urkud) [inline Mathlib/Analysis/Convex/Gauge.lean:549]:
```suggestion
```

**Review by ocfnash — APPROVED** [review]
Thanks!

bors merge

**Pull request closed without merging**